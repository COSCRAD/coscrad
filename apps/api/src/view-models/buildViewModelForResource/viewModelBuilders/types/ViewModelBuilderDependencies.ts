import { ConfigService } from '@nestjs/config';
import { RepositoryProvider } from '../../../../persistence/repositories/repository.provider';

export type ViewModelBuilderDependencies = {
    repositoryProvider: RepositoryProvider;
    configService: ConfigService;
};
