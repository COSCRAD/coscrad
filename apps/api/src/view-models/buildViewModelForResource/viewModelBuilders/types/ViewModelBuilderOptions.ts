export type ViewModelBuilderOptions = {
    shouldReturnUnpublishedEntities: boolean;
};

export const getDefaultViewModelBuilderOptions = (): ViewModelBuilderOptions => ({
    shouldReturnUnpublishedEntities: false,
});
