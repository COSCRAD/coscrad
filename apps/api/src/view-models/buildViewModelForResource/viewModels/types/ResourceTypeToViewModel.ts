import { BookViewModel } from '../book.view-model';
import { PhotographViewModel } from '../photograph.view-model';
import { SpatialFeatureViewModel } from '../spatial-data/spatial-feature.view-model';
import { TermViewModel } from '../term.view-model';
import { TranscribedAudioViewModel } from '../transcribed-audio/transcribed-audio.view-model';
import { VocabularyListViewModel } from '../vocabulary-list.view-model';

export type ResourceTypeToViewModel = {
    term: TermViewModel;
    vocabularyList: VocabularyListViewModel;
    transcribedAudio: TranscribedAudioViewModel;
    book: BookViewModel;
    photograph: PhotographViewModel;
    spatialFeature: SpatialFeatureViewModel;
};
