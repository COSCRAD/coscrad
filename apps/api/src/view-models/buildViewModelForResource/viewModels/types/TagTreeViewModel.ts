import { TagViewModel } from '../tag.view-model';

export type TagTreeViewModel = {
    label: string;

    nodes: TagTree[];
};

type TagTree = {
    tag: TagViewModel;

    children: TagTree[];
};

const foo: TagTreeViewModel = {
    label: 'tag tree',
    nodes: [
        {
            tag: {
                label: 'mammals',
                id: '1',
            },
            children: [
                {
                    tag: {
                        label: 'cats',
                        id: '3',
                    },
                    children: [
                        {
                            tag: {
                                label: 'big cats',
                                id: '35',
                            },
                            children: [],
                        },
                        {
                            tag: {
                                label: 'domesticated cats',
                                id: '1',
                            },
                            children: [],
                        },
                    ],
                },
                {
                    tag: {
                        label: 'dogs',
                        id: '7',
                    },
                    children: [],
                },
                {
                    tag: {
                        label: 'rodents',
                        id: '12',
                    },
                    children: [],
                },
            ],
        },
    ],
};

foo;
