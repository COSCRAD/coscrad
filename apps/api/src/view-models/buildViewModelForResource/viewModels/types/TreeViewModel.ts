import { EdgeConnection } from '../../../../domain/models/context/edge-connection.entity';
import { NoteViewModel } from '../../../edgeConnectionViewModels/note.view-model';
import { TagViewModel } from '../tag.view-model';

export type NoteTreeViewModel = {
    label: string;

    nodes: NoteTree[];
};

type NoteTree = {
    tag: TagViewModel;

    notes: NoteViewModel[];

    children: NoteTree[];
};

const foo: NoteTreeViewModel = {
    label: 'notes',
    nodes: [
        {
            tag: {
                label: 'mammals',
                id: '1',
            },
            notes: [new NoteViewModel({} as EdgeConnection)],
            children: [
                {
                    tag: {
                        label: 'cats',
                        id: '3',
                    },
                    notes: [new NoteViewModel({} as EdgeConnection)],
                    children: [
                        {
                            tag: {
                                label: 'big cats',
                                id: '35',
                            },
                            notes: [
                                new NoteViewModel({} as EdgeConnection),
                                new NoteViewModel({} as EdgeConnection),
                            ],
                            children: [],
                        },
                        {
                            tag: {
                                label: 'domesticated cats',
                                id: '1',
                            },
                            notes: [
                                new NoteViewModel({} as EdgeConnection),
                                new NoteViewModel({} as EdgeConnection),
                                new NoteViewModel({} as EdgeConnection),
                            ],
                            children: [],
                        },
                    ],
                },
                {
                    tag: {
                        label: 'dogs',
                        id: '7',
                    },
                    notes: [
                        new NoteViewModel({} as EdgeConnection),
                        new NoteViewModel({} as EdgeConnection),
                        new NoteViewModel({} as EdgeConnection),
                        new NoteViewModel({} as EdgeConnection),
                        new NoteViewModel({} as EdgeConnection),
                    ],
                    children: [],
                },
                {
                    tag: {
                        label: 'rodents',
                        id: '12',
                    },
                    notes: [
                        new NoteViewModel({} as EdgeConnection),
                        new NoteViewModel({} as EdgeConnection),
                    ],
                    children: [],
                },
            ],
        },
    ],
};

foo;
