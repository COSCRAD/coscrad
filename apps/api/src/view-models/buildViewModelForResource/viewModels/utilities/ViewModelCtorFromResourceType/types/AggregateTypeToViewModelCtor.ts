import { aggregateTypeToViewModelCtor } from '../aggregateTypeToViewModelCtor';

export type AggregateTypeToViewModelCtor = {
    [K in keyof typeof aggregateTypeToViewModelCtor]: InstanceType<
        typeof aggregateTypeToViewModelCtor[K]
    >;
};
