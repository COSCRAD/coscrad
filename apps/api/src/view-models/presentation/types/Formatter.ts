export type Formatter<T> = (dataToFormat: T) => string;
