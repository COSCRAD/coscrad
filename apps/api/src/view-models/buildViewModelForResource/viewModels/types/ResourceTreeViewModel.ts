import { Book } from '../../../../domain/models/book/entities/book.entity';
import { ResourceType, resourceTypes } from '../../../../domain/types/resourceTypes';
import { BaseViewModel } from '../base.view-model';
import { BookViewModel } from '../book.view-model';
import { TagViewModel } from '../tag.view-model';
import { ResourceTypeToViewModel } from './ResourceTypeToViewModel';

export type ResourceTreeViewModel<TResourceType extends ResourceType> = {
    resourceType: TResourceType;

    nodes: ResourceTree<ResourceTypeToViewModel[TResourceType]>[];
};

type ResourceTree<TViewModel extends BaseViewModel> = {
    tag: TagViewModel;

    resources: TViewModel[];

    children: ResourceTree<TViewModel>[];
};

const foo: ResourceTreeViewModel<typeof resourceTypes.book> = {
    resourceType: resourceTypes.book,
    nodes: [
        {
            tag: {
                label: 'mammals',
                id: '1',
            },
            resources: [new BookViewModel({} as Book)],
            children: [
                {
                    tag: {
                        label: 'cats',
                        id: '3',
                    },
                    resources: [new BookViewModel({} as Book)],
                    children: [
                        {
                            tag: {
                                label: 'big cats',
                                id: '35',
                            },
                            resources: [
                                new BookViewModel({} as Book),
                                new BookViewModel({} as Book),
                            ],
                            children: [],
                        },
                        {
                            tag: {
                                label: 'domesticated cats',
                                id: '1',
                            },
                            resources: [
                                new BookViewModel({} as Book),
                                new BookViewModel({} as Book),
                                new BookViewModel({} as Book),
                            ],
                            children: [],
                        },
                    ],
                },
                {
                    tag: {
                        label: 'dogs',
                        id: '7',
                    },
                    resources: [
                        new BookViewModel({} as Book),
                        new BookViewModel({} as Book),
                        new BookViewModel({} as Book),
                        new BookViewModel({} as Book),
                        new BookViewModel({} as Book),
                    ],
                    children: [],
                },
                {
                    tag: {
                        label: 'rodents',
                        id: '12',
                    },
                    resources: [new BookViewModel({} as Book), new BookViewModel({} as Book)],
                    children: [],
                },
            ],
        },
    ],
};

foo;
