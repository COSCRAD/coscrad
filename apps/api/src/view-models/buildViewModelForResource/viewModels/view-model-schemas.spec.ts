import { getCoscradDataSchema } from '@coscrad/data-types';
import { Term } from '../../../domain/models/term/entities/term.entity';
import { DTO } from '../../../types/DTO';
import { TermViewModel } from './term.view-model';

/***
 * Note: This is a temporary test. Once we expose the view model schemas
 * as part of a query, we can check a snapshot of part of said query's
 * integration test.
 */
describe(`Term View Model's schema`, () => {
    const _ = new Term({} as DTO<Term>);

    it('should match the snapshot', () => {
        const schema = getCoscradDataSchema(TermViewModel);

        expect(schema).toMatchSnapshot();
    });
});
