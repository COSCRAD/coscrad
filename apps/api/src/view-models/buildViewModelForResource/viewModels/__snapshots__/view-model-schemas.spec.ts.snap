// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Term View Model's schema should match the snapshot 1`] = `
Object {
  "audioURL": Object {
    "coscradDataType": "URL",
    "isArray": false,
    "isOptional": false,
  },
  "contributor": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "sourceProject": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": true,
  },
  "term": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": true,
  },
  "termEnglish": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": true,
  },
}
`;
